util/progress/progressui/colors.go

```diff
@@ -0,0 +1,60 @@
+package progressui
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/morikuni/aec"
+)
+
+var rex = regexp.MustCompile("(\\w+)=(\\w+)")
+
+var termColorMap = map[string]aec.ANSI{
+	"default": aec.DefaultF,
+
+	"black":   aec.BlackF,
+	"blue":    aec.BlueF,
+	"cyan":    aec.CyanF,
+	"green":   aec.GreenF,
+	"magenta": aec.MagentaF,
+	"orange":  aec.Color8BitF(aec.NewRGB8Bit(255, 140, 0)),
+	"red":     aec.RedF,
+	"white":   aec.WhiteF,
+	"yellow":  aec.YellowF,
+
+	"light-black":   aec.LightBlackF,
+	"light-blue":    aec.LightBlueF,
+	"light-cyan":    aec.LightCyanF,
+	"light-green":   aec.LightGreenF,
+	"light-magenta": aec.LightMagentaF,
+	"light-orange":  aec.Color8BitF(aec.NewRGB8Bit(255, 165, 0)),
+	"light-red":     aec.LightRedF,
+	"light-white":   aec.LightWhiteF,
+	"light-yellow":  aec.LightYellowF,
+}
+
+func setUserDefinedTermColors(colorsEnv string) {
+
+	data := rex.FindAllStringSubmatch(colorsEnv, -1)
```

Comment:
Not sure if regex parsing is the right way to split the data like this, since then we won't explicitly reject meaningless input.

We probably want something similar to how we do `--output` option handling, by using a csv parser, and then splitting each option by `=`. See https://github.com/moby/buildkit/blob/master/cmd/buildctl/build/output.go#L20-L31.

Reply:
I'll take a look at this and also supporting non-predefined colors. The current approach was largely to get something working that was simple and somewhat forgiving but overall agree with your thoughts here.

Reply:
Done